csvy/readers.py

```diff
@@ -62,6 +64,51 @@ def get_comment(line: str, marker: str = "---") -> str:
         return "" if line.startswith(marker) else line.split(marker)[0]
 
 
+def merge_csv_options_with_dialect(
```

Comment:
As far as I can tell, this is identical to the function in `writers.py`. I would put it in a helper `utils.py` module and have `readers.py` and `writers.py` import it.

Some tests will need moving to `test_utils.py`.